### ISSUE TITLE
Failure 0.1.2 Release

### ISSUE THREAD

**Issue by mitsuhiko** [rust-lang-nursery/failure#230]
I think master is ready for a release but I still need to get access to crates.io. I would love if people give the master version a try however since this library is light on tests currently.

**Comment by BurntSushi** [comment]
cc @alexcrichton

**Comment by alexcrichton** [comment]
Ok! @mitsuhiko you should be invited now and once accepting shoudl be a member of the @rust-lang-nursery/failure team, let me know if something is awry though!

**Comment by dekellum** [comment]
If anyone is in a similar circumstance with their *failure* dependency, please test like in dekellum/body-image@8118da3, per @mitsuhiko's [request](#issue-345232266). Thanks!

**Comment by mykmelez** [comment]
> I think master is ready for a release but I still need to get access to crates.io. I would love if people give the master version a try however since this library is light on tests currently.

It passes tests in the [rkv crate](https://github.com/mozilla/rkv) per [Appveyor](https://ci.appveyor.com/project/mykmelez/rkv-iy5s5/build/1.0.21) and [Travis CI](https://travis-ci.org/mykmelez/rkv/builds/409022116).

**Comment by withoutboats** [comment]
Thanks for your help @alexcrichton

**Comment by mitsuhiko** [comment]
This brings up the question if a point release should deprecate `cause()` I suppose. But the name is really bad so I at least want to bring in `as_fail()`.

**Comment by mitsuhiko** [comment]
I also want to introduce another deprecation through this: #231

That way users only have to suffer through the changes once.

**Comment by tismith** [comment]
I'm not sure a point release adding a deprecation is really in the spirit of semver, but either way - once the depreciation is taken care of, looks like https://github.com/tismith/exitfailure is passing ok -  https://github.com/tismith/exitfailure/pull/18 (appveyor still running).

**Comment by mitsuhiko** [comment]
I would love to not have that deprecation warning but I don’t know how to update everything to 0.2 without breaking the ecosystem otherwise.

**Comment by untitaker** [comment]
Not sure if intentional, but the backtrace feature is disabled by default in master while enabled by default in 0.1.1. This could be considered a breaking change.

**Comment by mitsuhiko** [comment]
@untitaker not intentional. Going to fix this.

**Comment by mmrath** [comment]
not able to use `context` on result. Is that an intentional breaking change?

**Comment by mitsuhiko** [comment]
@mmrath what do you mean by not being able to use context on result? The method still exists.

**Comment by mmrath** [comment]
@mitsuhiko Sample [repo](https://github.com/mmrath/fina-server/tree/failure-master)

When change version from 0.1 to git master I see an error like this. With 0.1.1 it compiles fine. other crates also have issues with the latest master i think. 

```
note: the method `context` exists but the following trait bounds were not satisfied:
           `std::result::Result<bool, argonautica::error::Error> : failure::result_ext::ResultExt<_, _>`
           `std::result::Result<bool, argonautica::error::Error> : failure::Fail`
           `&std::result::Result<bool, argonautica::error::Error> : failure::Fail`
           `&mut std::result::Result<bool, argonautica::error::Error> : failure::Fail`
   = help: items from traits can only be used if the trait is in scope
   = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
           `use failure::result_ext::ResultExt;`
```

**Comment by mitsuhiko** [comment]
That’s because you are not using the patch feature to replace it and you
now end up with two incompatible versions of failure in the same app.

**Comment by mmrath** [comment]
would you know how do i fix it?

**Comment by mitsuhiko** [comment]
Use the patch feature in cargo instead of using a git dependency directly.
https://doc.rust-lang.org/cargo/reference/manifest.html

**Comment by kyren** [comment]
Not sure if this is the correct place for this discussion, but with commit 3d5b8f9172579fc14ac748190e0f4c8fc6103fe2 does this mean that it will be impossible to use `std` features in `failure` without enabling backtraces?  If so, that's much less useful than them being controllable separately.

**Comment by kyren** [comment]
@mitsuhiko Yeah, I'm aware it was the behavior in previous released versions.  Disabling `backtrace` was actually the specific feature I was waiting on 1.0 for (but obviously plans have changed since then).

I don't remember where the discussion was, but iirc it was a planned 1.0 feature?  I can try and find the original discussion, but it isn't super critical to me anymore anyway.  If you would rather NOT have this behavior in the next release that's fine as well, but if it's easy to do, it would probably be useful?

If this thread is for discussing backwards compatibility though, this is probably wildly off-topic.

**Comment by mitsuhiko** [comment]
I just can’t change this in a minor release

**Comment by dekellum** [comment]
An 0.2 release is still possible, and sufficient from a semver perspective, for @kyren request, right?

**Comment by kyren** [comment]
Ohhh, I didn't think about this enough.  I was thinking that it would be sufficient to make the backtrace feature enabled by default but *not* via std, but I suppose form a semver perspective it would have to be enabled when std is enabled as well?  It's not an API breaking change, and you'd have to already have disabled default features *and* be opting into std, but I suppose that IS still strictly a breaking change, darn.  I hope I'm thinking about this correctly, if I'm missing some other obvious semver issue let me know, but I understand if even that one is sufficiently breaking.

Yeah, okay, I guess keep it in mind for a future 0.2 or 1.0 but otherwise sorry for the noise!

**Comment by withoutboats** [comment]
My understanding is that once rust-lang/rfcs#2504 is landed and stabilized, failure will upgrade so that users are expected to implement `std::error::Error` instead of `failure::Fail`. That will be a breaking change, but once it happens it frees failure from worrying about defining the shared interface used by all other libraries, and failure will be more able to make breaking changes as it evolves from then on.

**Comment by mitsuhiko** [comment]
I am going to release current master tomorrow. Last chance to give feedback.

With regards to std implying backtrace: i will have a look next release if we can do something there. In the ideal case the std feature does not actually do anything anyways.

**Comment by mitsuhiko** [comment]
This has been released.

**Comment by cauebs** [comment]
I'm sorry, I'm a bit out of the loop. What are the changes in this version?

**Comment by otavio** [comment]
@cauebs there was #241 adding the changes to the release notes. I hope it helps you.
